flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionTest.java

```diff
@@ -540,6 +557,65 @@ public void testSlotReleaseAtomicallyReleasesExecution() throws Exception {
 		});
 	}
 
+	/**
+	 * Tests that producedPartitions are registered synchronously under an asynchronous interface.
+	 */
+	@Test(expected = IllegalStateException.class)
+	public void testSynchronousRegisterProducedPartitions() throws Exception {
+		ExecutionJobVertex jobVertex = getExecutionVertex(
+				new JobVertexID(),
+				new DirectScheduledExecutorService(),
+				EAGER);
+		IntermediateResult result =
+				new IntermediateResult(new IntermediateDataSetID(), jobVertex, 1, ResultPartitionType.PIPELINED);
+		ExecutionVertex vertex =
+				new ExecutionVertex(jobVertex, 0, new IntermediateResult[]{result}, Time.minutes(1));
+		ExecutionEdge edge = createMockExecutionEdge(1);
+		result.getPartitions()[0].addConsumerGroup();
+		result.getPartitions()[0].addConsumer(edge, 0);
```

Comment:
Are these lines necessary?

Reply:
Yep, this part is needed, there has to be one consumer.